lib/codecs/src/encoding/mod.rs

```diff
@@ -362,12 +362,14 @@ impl SerializerConfig {
                 FramingConfig::LengthDelimited(LengthDelimitedEncoderConfig::default())
             }
             SerializerConfig::Csv(_)
-            | SerializerConfig::Gelf
             | SerializerConfig::Json(_)
             | SerializerConfig::Logfmt
             | SerializerConfig::NativeJson
             | SerializerConfig::RawMessage
             | SerializerConfig::Text(_) => FramingConfig::NewlineDelimited,
+            SerializerConfig::Gelf => {
```

Comment:
In order to be consistent with the `default_stream_framing` of the `DeserializerConfig`. Noticed that this change was missing in #18008. Changed it here, but tell me if you want this PR to not do anything out of scope from #20768.

Reply:
Huh, yeah, this does seem like an oversight. I'm trying to understand the impact of it though since the original PR did seem to resolve the issue for people 🤔

Reply:
It seems that the changed struct was `EncodingConfigWithFraming` (#18419).

It seems that a few places relies on `EncodingConfigWithFraming` to get this default behaviour, and other places uses this method `SerializerConfig::default_stream_framing`. For example, see this line https://github.com/vectordotdev/vector/blob/67a5e4682138721575b22369cf5ba629cd86ce55/src/components/validation/resources/mod.rs#L113
Where we can see its relying on `SerializerConfig::default_stream_framing` in order to get the default framing for a given encoder.

There are a lot of sinks that relies on `EncodingConfigWithFraming` having a Option<FramingConfig> in order to calculate that default (https://github.com/search?q=repo%3Avectordotdev%2Fvector%20EncodingConfigWithFraming&type=code) https://github.com/vectordotdev/vector/blob/67a5e4682138721575b22369cf5ba629cd86ce55/src/codecs/encoding/config.rs#L110

Reply:
Aha, gotcha, thanks for looking into that.